identity/src/main/java/com/android/identity/credential/AuthenticationKey.kt

```diff
@@ -0,0 +1,229 @@
+    val attestation: List<X509Certificate>
+        get() {
+            return secureArea.getKeyInfo(alias).attestation
+        }
+
+    /**
+     * The secure area for the authentication key.
+     *
+     * This can be used together with the alias returned by [alias].
+     *
+     * @return The [SecureArea] used.
+     */
+    lateinit var secureArea: SecureArea
+
+    private lateinit var credential: Credential
+
+    internal var replacementAlias: String? = null
+
+    /**
+     * Deletes the authentication key.
+     *
+     *
+     * After deletion, this object should no longer be used.
+     */
+    fun delete() {
+        secureArea.deleteKey(alias)
+        credential.removeAuthenticationKey(this)
+    }
+
+    /**
+     * Increases usage count of the authentication key.
+     */
+    fun increaseUsageCount() {
+        usageCount += 1
+        credential.saveCredential()
+    }
+
+    fun toCbor(): DataItem {
+        val builder = CborBuilder()
+        val mapBuilder: MapBuilder<CborBuilder> = builder.addMap()
+        mapBuilder.put("alias", alias)
+            .put("domain", domain)
+            .put("secureAreaIdentifier", secureArea.identifier)
+            .put("usageCount", usageCount.toLong())
+            .put("data", issuerProvidedData)
+            .put("validFrom", validFrom.toEpochMilli())
+            .put("validUntil", validUntil.toEpochMilli())
+            .put("applicationData", privateApplicationData.encodeAsCbor())
+            .put("authenticationKeyCounter", authenticationKeyCounter)
+        if (replacementAlias != null) {
+            mapBuilder.put("replacementAlias", replacementAlias)
+        }
+        return builder.build()[0]
+    }
+
+    /**
+     * The pending auth key that will replace this key once certified or `null` if no
+     * key is designated to replace this key.
+     */
+    val replacement: PendingAuthenticationKey?
+        get() {
+            if (replacementAlias == null) {
+                return null
+            }
+            for (pendingAuthKey in credential.pendingAuthenticationKeys) {
+                if (pendingAuthKey.alias == replacementAlias) {
+                    return pendingAuthKey
+                }
+            }
+            Logger.w(TAG, "Pending key with alias $replacementAlias which " +
+                    "is intended to replace this key does not exist")
+            return null
+        }
+
+    fun setReplacementAlias(alias: String) {
+        replacementAlias = alias
+        credential.saveCredential()
+    }
+
+    companion object {
+        const val TAG = "AuthenticationKey"
+
+        fun create(
+            pendingAuthenticationKey: PendingAuthenticationKey,
+            issuerProvidedAuthenticationData: ByteArray,
+            validFrom: Timestamp,
+            validUntil: Timestamp,
+            credential: Credential
+        ): AuthenticationKey {
+            val ret = AuthenticationKey()
+            ret.alias = pendingAuthenticationKey.alias
```

Comment:
using the `.apply{ .. }` [scope function](https://kotlinlang.org/docs/scope-functions.html#apply) we can make it neater 

```
        fun create(
            pendingAuthenticationKey: PendingAuthenticationKey,
            issuerProvidedAuthenticationData: ByteArray,
            validFrom: Timestamp,
            validUntil: Timestamp,
            credential: Credential
        ) = AuthenticationKey().apply {
            alias = pendingAuthenticationKey.alias
            domain = pendingAuthenticationKey.domain
            issuerProvidedData = issuerProvidedAuthenticationData
            this.validFrom = validFrom
            this.validUntil = validUntil
            this.credential = credential
            secureArea = pendingAuthenticationKey.secureArea
            privateApplicationData = pendingAuthenticationKey.privateApplicationData
            authenticationKeyCounter = pendingAuthenticationKey.authenticationKeyCounter
        }
```